business/istio_config.go

```diff
@@ -566,14 +564,21 @@ func (in *IstioConfigService) hasMeshPolicyEnabled(namespaces []string) (bool, e
 
 		for _, peer := range peers.([]interface{}) {
 			peerMap := peer.(map[string]interface{})
-			if _, present := peerMap["mtls"]; present {
-				mtlsEnabled = true
-				break
+			if mtls, present := peerMap["mtls"]; present {
+				if mtlsMap, ok := mtls.(map[string]interface{}); ok {
+					// mTLS enabled in case there is an empty map or mode is STRICT
+					if mode, found := mtlsMap["mode"]; !found || mode == "STRICT" {
+						return true, nil
+					}
+				} else {
+					// mTLS enabled in case mtls object is empty
+					return true, nil
+				}
```

Comment:
I'm not sure if it applies to a real-world case, but shouldn't be there another `else` case to deal with `present` being false? It seems to me that the comment above ( `// mTLS enabled in case mtls object is empty`) would also apply to that 'else' condition, no?

Reply:
( @xeviknal tell me, I might be wrong :) )